lib/util/lifted/influx/coordinator/statement_executor.go

```diff
@@ -1402,6 +1407,130 @@ func (e *StatementExecutor) executeShowMeasurementsStatement(q *influxql.ShowMea
 	}, seq)
 }
 
+func (e *StatementExecutor) executeShowMeasurementsDetailStatement(stmt *influxql.ShowMeasurementsDetailStatement, ctx *query.ExecutionContext, seq int) error {
+	if stmt.Database == "" {
+		return coordinator.ErrDatabaseNameRequired
+	}
+	var mms influxql.Measurements
+	if stmt.Source != nil {
+		mms = influxql.Measurements{stmt.Source.(*influxql.Measurement)}
+	}
+
+	measurements, err := e.MetaClient.MatchMeasurements(stmt.Database, mms)
+	if err != nil {
+		return err
+	}
+
+	var blank2Nil = func(s string) string {
+		// not return a empty string, return "<nil>" instead
+		if s == "" {
+			return "<nil>"
+		}
+		return s
+	}
+
+	// sliceFirstColumn gets all shard keys and indexes from the returned models.Row.Values array, then returns a sorted slice.
+	// sk: true if the values are shard keys.
+	var sliceFirstColumn = func(values [][]interface{}, sk bool) (ret []string) {
+		ret = make([]string, 0, len(values))
+		if sk {
+			for i := range values {
+				ret = append(ret, values[i][0].([]string)...)
+			}
+		} else {
+			for i := range values {
+				ret = append(ret, values[i][0].(string))
+			}
+		}
+		sort.Strings(ret)
+		return
+	}
+	var emitted = false
+	for key, m := range measurements {
+		originName := m.OriginName()
+		row := &models.Row{Name: originName, Tags: make(map[string]string, 1), Columns: []string{"Detail"}}
+		values := make([][]interface{}, 0, 8)
```

Comment:
confusing magic number `8`, can you give me a brief description?

Reply:
The `values` here, which represents the detailed schema information of measurements, has 8 rows at most, including shard keys, indexes, engine type, sort keys, primary keys, compaction type, field keys and tag keys.

Reply:
After the policy information was added, now it becomes 9. I also leave a code comment here to improve its readability.